lib/charms/layer/jenkins/packages.py

```diff
@@ -72,7 +86,17 @@ def _install_from_remote_deb(self, url):
         hookenv.log("Getting remote jenkins package: %s" % url)
         tempdir = tempfile.mkdtemp()
         target = os.path.join(tempdir, 'jenkins.deb')
-        subprocess.check_call(("wget", "-q", "-O", target, url))
+
+        try:
+            proxy_env = _juju_proxy_env()
+            if len(proxy_env) == 0:
+                # If there are no juju proxy variables set,
+                # abort and test w/ unmodified environment
+                raise NoProxyError
+            subprocess.check_call(("wget", "-q", "-O", target, url),
+                                  env={**os.environ, **proxy_env})
+        except (subprocess.CalledProcessError, NoProxyError):
+            subprocess.check_call(("wget", "-q", "-O", target, url))
```

Comment:
I think this could be rewritten a tad more clearly like this:

```python
proxy_env = _juju_proxy_env()
try:
    subprocess.check_call(("wget", "-q", "-O", target, url),
        env={**os.environ, **proxy_env})
except subprocess.CalledProcessError:
    # If there are juju proxy variables set, try also with unmodified environment
    if len(proxy_env) > 0:
        subprocess.check_call(("wget", "-q", "-O", target, url))
    else:
        raise
```

This avoids the need to introduce ```NoProxyError```.

Reply:
I like that better, wasn't familiar with the bare raise syntax. Though, I think I'll flip the logic in the except to get rid of the else. I'll push after a re-test...